c/src/ml-api-service-remote.c

```diff
@@ -121,32 +121,35 @@ _mlrs_get_conn_type (const gchar * value)
  * @brief Get edge info from ml_option.
  */
 static void
-_mlrs_get_edge_info (ml_option_h option, edge_info_s * edge_info)
+_mlrs_get_edge_info (ml_option_h option, edge_info_s ** edge_info)
 {
+  edge_info_s *_info;
   void *value;
 
-  if (ML_ERROR_NONE == ml_option_get (option, "host", &value)) {
-    g_free (edge_info->host);
-    edge_info->host = g_strdup (value);
-  }
+  *edge_info = _info = g_new0 (edge_info_s, 1);
+
+  if (ML_ERROR_NONE == ml_option_get (option, "host", &value))
+    _info->host = g_strdup (value);
+  else
+    _info->host = g_strdup ("localhost");
   if (ML_ERROR_NONE == ml_option_get (option, "port", &value))
-    edge_info->port = *((guint *) value);
-  if (ML_ERROR_NONE == ml_option_get (option, "dest-host", &value)) {
-    g_free (edge_info->dest_host);
-    edge_info->dest_host = g_strdup (value);
-  }
+    _info->port = *((guint *) value);
+  if (ML_ERROR_NONE == ml_option_get (option, "dest-host", &value))
+    _info->dest_host = g_strdup (value);
+  else
+    _info->dest_host = g_strdup ("localhost");
   if (ML_ERROR_NONE == ml_option_get (option, "dest-port", &value))
-    edge_info->dest_port = *((guint *) value);
+    _info->dest_port = *((guint *) value);
   if (ML_ERROR_NONE == ml_option_get (option, "connect-type", &value))
-    edge_info->conn_type = _mlrs_get_conn_type (value);
+    _info->conn_type = _mlrs_get_conn_type (value);
+  else
+    _info->conn_type = NNS_EDGE_CONNECT_TYPE_UNKNOWN;
```

Comment:
Can NNS_EDGE_CONNECT_TYPE_UNKNOWN  work with host and dest_host those set 'localhost '?
How about setting the default connection type?

Reply:
Now user should set the connection type. If it is not set (unknown type), this function returns error.

Reply:
Thank you.